alf/environments/parallel_environment.py

```diff
@@ -155,22 +183,69 @@ def _step(self, actions):
         Returns:
             Batch of observations, rewards, and done flags.
         """
-        time_steps = [
-            env.step(action, self._blocking)
-            for env, action in zip(self._envs, self._unstack_actions(actions))
-        ]
+        if self._spare_queue:
+            actions = self._unstack_actions(actions)
+            time_steps = self._handle_last_done(actions)
```

Comment:
The name "handle_last_done" is confusing. In addition to handle last done, it also performs step.

Reply:
Changed to _step_or_handle_last_done